update resolver parameters

bugfood opened the pull request:
This fixes two separate issues with the `resolver` defined type:
1. The `accepted_payload_size` puppet parameter must be specified, despite HAProxy itself allowing the corresponding option to be unspecified.
2. The `parse-resolv-conf` HAProxy option is not supported.

I have updated the tests accordingly.

david22swan commented on manifests/resolver.pp:
```diff
@@ -87,11 +96,13 @@
   $timeout                 = undef,
-  $accepted_payload_size   = undef,
+  # https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#5.3.2-accepted_payload_size
+  Optional[Integer[512, 8192]] $accepted_payload_size = undef,
```
Could you align the different variable lengths

bugfood replied:
Sure, done; I did this as an additional commit rather than mix functional and whitespace changes together.

david22swan commented:
@bugfood One small change is needed, otherwise this looks good

jordanbreen28 commented:
Hi @bugfood - this change looks good. Can you address @david22swan's comment above, and rebase with the current main, so we can proceed with this. 
Thanks

bugfood commented:
Thanks, yes, I will take a look at this.

bugfood commented:
@jordanbreen28 @david22swan thanks, I amended the PR for:
* whitespace change
* rebase

I've been using my branch on live hosts since I made the PR, and it's been fine so far, at least for my use case. The rebased branch works as well.

jordanbreen28 commented:
@bugfood thank you for all your work on this so far!
There seems to have been some test failrues - can you take a look at these?

bugfood commented:
> @bugfood thank you for all your work on this so far! There seems to have been some test failures - can you take a look at these?

Yes, rubocop seems to be getting stricter over time. I fixed that failure, and I don't see any other failures in the test report. Hopefully the test suite will pass now.

bugfood commented:
Note that there is a deprecation warning on `pdk bundle exec rake spec`:
```
Deprecation Warnings:

Using `should` from rspec-expectations' old `:should` syntax without explicitly enabling the syntax is deprecated. Use the new `:expect` syntax or explicitly enable `:should` with `config.expect_with(:rspec) { |c| c.syntax = :should }` instead. Called from /home/chickey/puppetlabs-haproxy/spec/classes/haproxy_spec.rb:28:in `block (5 levels) in <top (required)>'.
```

This warns on the test I added, but that's only because the new test is listed first in the file. The new tests I added follow the same construction as the other tests, and the same warning happens on current upstream `main`.

I don't think that the pre-existing warning needs to be fixed here, but if so, I would need some further guidance on that. The string "should" does not appear in that file; I presume that some higher-level construct is using ":should", but I'm not familiar enough with rspec to know what would need to be changed. Other examples of that error I can find by a web search are for code that looks rather different.

jordanbreen28 commented:
Hi again @bugfood :-) 
Great work on this one, theres one failure but as before I expect it to be transient. 
Let me re-kick the test here so hopefully we can proceed!

jordanbreen28 reviewed (approved):
LGTM!

Thanks for your work on this & for bringing the deprecation warning to our attention.
